cmd/argocd/commands/app.go

```diff
@@ -1049,24 +1051,55 @@ func NewApplicationDeleteCommand(clientOpts *argocdclient.ClientOptions) *cobra.
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
+
 			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
 			defer util.Close(conn)
+
 			for _, appName := range args {
 				appDeleteReq := applicationpkg.ApplicationDeleteRequest{
 					Name: &appName,
 				}
 				if c.Flag("cascade").Changed {
 					appDeleteReq.Cascade = &cascade
 				}
+
 				_, err := appIf.Delete(context.Background(), &appDeleteReq)
 				errors.CheckError(err)
+
+				if c.Flag("wait").Changed {
+					err = waitForAppDeletion(appIf, appName, timeout)
+					errors.CheckError(err)
+				}
 			}
 		},
 	}
 	command.Flags().BoolVar(&cascade, "cascade", true, "Perform a cascaded deletion of all application resources")
+	command.Flags().BoolVar(&wait, "wait", false, "Will cause the command to only return once the application has been deleted")
+	command.Flags().IntVar(&timeout, "timeout", 60, "Time to spend waiting for application to delete, default 60s")
+
 	return command
 }
 
+func waitForAppDeletion(c applicationpkg.ApplicationServiceClient, appName string, timeout int) error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(timeout)*time.Second)
+	defer cancel()
+
+	for {
+		_, err := c.Get(context.Background(), &applicationpkg.ApplicationQuery{Name: &appName})
+		if err != nil {
+			if strings.HasSuffix(err.Error(), "not found") {
+				return nil
+			}
+		}
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("wait timed out after %ds waiting for app %s to be deleted", timeout, appName)
+		default:
+			time.Sleep(1 * time.Second)
```

Comment:
Can you please check this polling interval is consistent with `app create`?

Reply:
Should we use `waitOnApplicationStatus` instead of creating new waiting logic here?

We use `waitOnApplicationStatus` for `argocd app {wait, sync, rollback}` commands and it already has much of the logic implemented here.

Reply:
@simster7, I have switched over to using `waitOnApplicationStatus`. I had to add a bit more logic. The events were also a bit misleading as they were saying created when the resources were actually being deleted.